### PR TITLE
Feature/new repo layout

### PR DESCRIPTION
**Pull request by pcrespov**
I would like to reshape the base structure of the repository to tackle better its purpose. 

The main objective is to end up with a repo full of demos and proof-of-concepts towards SIM-Core. All the code and notes produced here will be the base for the [first milestone](https://github.com/orgs/ITISFoundation/projects/1#card-5505425) at the end of the year.

Once reached that point, the development of the SIM-Core framework can be done in a separate and dedicated repository.

### CODE REVIEW

File: docs/coding-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Coding Guidelines
+
+The code on this project should follow these guidelines
+
+ - C++: https://google.github.io/styleguide/cppguide.html
+ - Python: https://www.python.org/dev/peps/pep-0008/
+ - JavaScript: https://github.com/airbnb/javascript
```
Comment (oetiker) [inline docs/coding-guidelines.md]: good choice :-)

**Comment by oetiker** [comment]
note that you can set the repo to 'block' merging until there is a a postitive review and no negative ones ... https://github.com/ITISFoundation/oSPARC_Test/settings/branches

also you can setup a https://github.com/qooxdoo/qooxdoo/blob/master/.github/CODEOWNERS to assign reviewers automatically ... this here is very simple ... it could even be structured to have different reviewers for different parts of the code

**Comment by oetiker** [comment]
if you hookup travis to your repo (it might make sense to actually split this one into multiple to have really one project per repo) then you can build on every PR and get an automatic review 

for inspiration check out

https://github.com/qooxdoo/qooxdoo/.travis.yml

**Comment by oetiker** [comment]
oops wrong link https://github.com/qooxdoo/qooxdoo/blob/master/.travis.yml

**Comment by pcrespov** [comment]
Yes, we already created a (card)[https://github.com/ITISFoundation/oSPARC_Test/projects/1#card-5556489]. CircleCI seems to suits our needs since it also includes C++ and as far as I understood it accepts containers.

**Pull request merged**